Projects/LinkedListIterator/LinkedListIterator/LinkedListIterator/SLList.h

```diff
@@ -0,0 +1,374 @@
+		Size = 0;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//								TEMPLATES & PREDICATES
+	//-------------------------------------------------------------------------------------------
+	template<typename Pred>
+	void ForEach(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			Predicate(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	template<typename Pred>
+	T& FindByPredicate(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			if (Predicate(Current->Element))
+			{
+				return &Current->Element;
+			}
+			Current = Current->Next;
+		}
+		return nullptr;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	template<typename Pred>
+	TSLList<T> FilterByPredicate(const Pred& Predicate)
+	{
+		TSLList<T> FilteredList;
+		FNode* Current = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			if (Predicate(Current->Element))
+			{
+				FilteredList.AddTail(Current->Element);
+			}
+			Current = Current->Next;
+		}
+		return FilteredList;
+	}
+
+	//-------------------------------------------------------------------------------------------
+	template<typename Pred>
+	void RemoveAllByPredicate(const Pred& Predicate)
+	{
+		FNode* Current = Head;
+		for (int i = 0; i < Size; i++)
+		{
+			T Element = Current->Element;
+			Current = Current->Next;
+
+			if (Predicate(Current->Element))
+			{
+				Remove(i);
+				i--;
+			}
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	//										ITERATOR
+	//-------------------------------------------------------------------------------------------
+	class FIterator
+	{
+	private:
+
+		FNode* Current = nullptr;
+
+	public:
+
+		FIterator(FNode* InCurrent)
+		{
+			Current = InCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		T& operator*()
+		{
+			return Current->Element;
+		}
+
+		const T& operator*() const
+		{
+			return Current->Element;
+		}
+
+		bool operator!=(FIterator Other)
+		{
+			return Current != Other.Current;
+		}
+
+		const bool operator!=(FIterator Other) const
+		{
+			return Current != Other.Current;
+		}
+
+	};
+
+	//-------------------------------------------------------------------------------------------
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
+	//-------------------------------------------------------------------------------------------
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+
+
+
+private:
+
+	//-------------------------------------------------------------------------------------------
+	//										UTILITY FUNCTIONS
+	//-------------------------------------------------------------------------------------------
+	FNode* GetNode(const int Index)
```

Comment:
Great implementation of this utility feature

Reply:
you could start to implement the iterations with this function.